frontend/packages/console-shared/src/components/catalog/utils/catalog-utils.tsx

```diff
@@ -155,6 +158,20 @@ export const isCatalogTypeEnabled = (catalogType: string): boolean => {
   return true;
 };
 
+export const useClusterSamplesOperatorEnabled = (): boolean => {
+  const [model] = useK8sModel({
+    group: 'samples.operator.openshift.io',
+    version: 'v1',
+    kind: 'Config',
+  });
+  const [resource, loaded] = useK8sGet<K8sResourceKind>(model, 'cluster');
```

Comment:
When the "Upload JAR file" feature depends on a Java builder image, we should check for that instead of checking that the samples operator is installed. That's correct for default installations, but customers might also install the builder images on their own os with another operator. And in this case, we want to show its option again.

So please update the check to a smaller but more accurate image stream check. This check then can hopefully live also in the dev-console package.